### PR TITLE
[RSDK-10176] - Add default camera behavior to object-tracker

### PR DESCRIPTION
**Pull request by jgeaso1266**
## Changes
- [x] allow `CameraName` to be an empty string in all `*FromCamera` methods
- [x] add tests to validate this behavior

### CODE REVIEW

File: object_tracker/object_tracker.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func newTracker(ctx context.Context, deps resource.Dependencies, conf resource.C
 		t.run(stream, t.cancelContext)
 	}, func() {
 		t.cancelFunc()
-		stream.Close(t.cancelContext)
+		viamutils.UncheckedError(stream.Close(t.cancelContext))
```
Comment (kharijarrett) [inline object_tracker/object_tracker.go]: Okay looked this up and I'm actually not sure if we want to do this (I might be persuaded).  Our docs specifically say not to use this to close writers ( https://pkg.go.dev/go.viam.com/utils#UncheckedErrorFunc).  The real question is how broadly is "writer" defined? Like the stream is "writing" pixel data.  

So I'm not feeling this one (if it ain't broke..) but again I can be convinced.

File: object_tracker/object_tracker_test.go [hunk 2]
```diff
@@ -340,3 +341,37 @@ func TestTracker(t *testing.T) {
...
+	fakeTracker := &myTracker{
+		camName: "test",
+		cancelContext: ctx,
+	}
+
+	invalidName := "not-camera"
+	invalidNameErrorMessage := "Camera name given to method, not-camera is not the same as configured camera test"
+
+	// Test invalid camera name in DetectionsFromCamera
+	_, err := fakeTracker.DetectionsFromCamera(ctx, invalidName, make(map[string]interface{}))
```
Comment (kharijarrett) [inline object_tracker/object_tracker_test.go]: And maybe make this empty map once, call it emptyMap, and reuse it.  This one I don't _actually_ care about.  Just a thought.

File: object_tracker/object_tracker_test.go [hunk 3]
```diff
@@ -340,3 +341,37 @@ func TestTracker(t *testing.T) {
...
+
+	// Test empty (valid) camera name in ClassificationsFromCamera
+	_, err = fakeTracker.ClassificationsFromCamera(ctx, "", 0, make(map[string]interface{}))
+	test.That(t, err, test.ShouldBeNil)
+
+	// Test invalid camera name in CaptureAllFromCamera
+	_, err = fakeTracker.CaptureAllFromCamera(ctx, invalidName, viscapture.CaptureOptions{ReturnImage: true}, make(map[string]interface{}))
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, invalidNameErrorMessage)
+}
```
Comment (kharijarrett) [inline object_tracker/object_tracker_test.go:387]: Also test out the empty cam name for CAFC.  If it's gone because you get other errors, you can just test that the error doesn't contain that invalid error substring.

File: object_tracker/object_tracker.go (hunk 1 above)
Comment (jgeaso1266) [reply object_tracker/object_tracker.go]: This is currently failing lint (at least locally) without this change. I'm not really sure how this passed before. I can try to reverse this change and see if it passes lint in Github actions.

File: object_tracker/object_tracker_test.go (hunk 3 above)
Comment (jgeaso1266) [reply object_tracker/object_tracker_test.go:387]: I was getting some seg faults because I don't load `t.currImg`. Maybe we could chat for a sec today to figure out how to do that.

**Review by kharijarrett — APPROVED** [review]
LGTM!
